integrationconnectors/advanced/main.tf

```diff
@@ -0,0 +1,251 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+data "google_project" "test_project" {
+}
+
+resource "google_secret_manager_secret" "secret-basic" {
+  secret_id = "<%= ctx[:vars]['secret_id'] %>"
+  replication {
+    user_managed {
+      replicas {
+        location = "us-central1"
+      }
+    }
+  }
+}
+
+
+resource "google_secret_manager_secret_version" "secret-version-basic" {
+  secret      = google_secret_manager_secret.secret-basic.id
+  secret_data = "dummypassword"
+}
+
+resource "google_secret_manager_secret_iam_member" "secret_iam" {
+  secret_id  = google_secret_manager_secret.secret-basic.id
+  role       = "roles/secretmanager.admin"
+  member     = "serviceAccount:${data.google_project.test_project.number}-compute@developer.gserviceaccount.com"
+  depends_on = [google_secret_manager_secret_version.secret-version-basic]
+}
+
+
+# [START integration_connectors_example]
+resource "google_integration_connectors_connection" "testconnection" {
+  name              = "test-connection"
+  description       = "tf updated description"
+  location          = "us-central1"
+  service_account   = "${data.google_project.test_project.number}-compute@developer.gserviceaccount.com"
+  connector_version = "projects/${data.google_project.test_project.project_id}/locations/global/providers/zendesk/connectors/zendesk/versions/1"
+  config_variable {
+    key           = "proxy_enabled"
+    boolean_value = false
+  }
+  config_variable {
+    key           = "sample_integer_value"
+    integer_value = 1
+  }
+
+  config_variable {
+    key = "sample_encryption_key_value"
+    encryption_key_value {
+      type         = "GOOGLE_MANAGED"
+      kms_key_name = "sampleKMSKkey"
+    }
+  }
+
+  config_variable {
+    key = "sample_secret_value"
+    secret_value {
+      secret_version = google_secret_manager_secret_version.secret-version-basic.name
+    }
+  }
+
+  suspended = false
+  auth_config {
+    additional_variable {
+      key          = "sample_string"
```

Comment:
These looks like simple key-value pairs. Does it help to have some many?

Reply:
I notice a repetition of this pattern. Please fix or add necessary code comments.